demos/python_demos/common/performance_metrics.py

```diff
@@ -0,0 +1,97 @@
+"""
+ Copyright (C) 2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from time import perf_counter
+import cv2
+from helpers import put_highlighted_text
+
+
+class Statistic:
+    def __init__(self):
+        self.latency = 0.0
+        self.period = 0.0
+        self.frame_count = 0
+
+
+class PerformanceMetrics:
+    def __init__(self, time_window=1.0):    # defines the length of the timespan used for 'current fps' calculation,
+        self.time_window_size = time_window # set to '1 second' by default
+        self.last_moving_statistic = Statistic()
+        self.current_moving_statistic = Statistic()
+        self.total_statistic = Statistic()
+        self.last_update_time = None
+    
+    def update(self, last_request_start_time, frame, position=(15, 30),
+               font_scale=0.75, color=(200, 10, 10), thickness=2):
+        current_time = perf_counter()
+        if self.last_update_time is None:
```

Comment:
Perhaps it would be better to initialize `last_update_time` in the constructor?

Reply:
I think it's better when the length of the period does not depend on when PerformanceMetrics object was initialized
https://github.com/opencv/open_model_zoo/pull/1222#discussion_r440455123

Reply:
OK, but I think the code could be rearranged to make the logic clearer. It seems to me that `self.last_update_time is None` is effectively equivalent to `self.total_statistic.frame_count == 0`, so the `self.last_update_time` initialization can be moved to the `else` branch of the `if self.total_statistic.frame_count != 0` statement.